### ISSUE TITLE
next/link doesn't work with custom server routing.

### ISSUE THREAD

**Issue by KhaledEmaraDev** [zeit/next.js#9288]
# Bug report

## Describe the bug

I have set up a simple example for this issue [https://github.com/KhaledEmaraDev/next-link-issue](here).

Basically, I have a `/` page with a link to `/reviews/:product` page, it sends the query parameter `product`. In the `/reviews` page I make a request to `/api/reviews` with the query parameter `product` using `isomorphic-unfetch` in `getInitialProps()` and return the result to the page component, which displays said result. The `/api/reviews` route reads the query parameter `product` and returns it back.

I have set up, a server, that parses the query and renders the `reviews` page. The code is exactly as that in `custom-express-server` example. I also made sure it follows the documentation.

On clicking the link on Chrome Dev 80.0.3955.3 on Android the page takes one minute or so before loading. If, however, I'm using a production build; the page instantly yields "An unexpected error has occurred". On reloading the page, it's displayed normally.

Logging the requests URLs in the express server: I found that on clicking the link it makes a request for this URL: /_next/static/<hash>/pages/reviews.js, immediately followed by a request to the error page: /_next/static/<hash>/pages/_error.js. Both of theses pages exist.

Note: This only happens on mobile. On desktop it works fine, even with mobile emulation.
Visiting the page normally without `next/link` also works, even on mobile.
I also tested with the stable version of Chrome on another version of Android, which exhibits the same behavior.

On changing the API URL to something that not on the same node server, it also works. I tested against an already published API that is an exact replica of this API, and it works just fine.

Testing the API page alone, without the page, also works fine.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. `git clone https://github.com/KhaledEmaraDev/next-link-issue.git`
2. `yarn install`
3. `yarn build`
4. `yarn start`
5. Visit [http://localhost:3000](http://localhost:3000) on mobile
6. Click on the "Reviews" link. "An Unexpected error has occurred" should appear.
7. Reload the page. It works fine.
8. Visit [http://localhost:3000/api/reviews?product=test](http://localhost:3000/api/reviews?product=test). Again, works fine.
9. Visit on Desktop. Works fine.
10. Visit the reviews page directly http://localhost:3000/reviews/test. Works fine.
11. Stop the server then start a dev one `yarn dev`.
12. Repeat step number 5. The pages takes one minute before rendering.

## System information

- OS: Linux, Android (Testing)
- Browser: Chrome Dev 80.0.3955.3 (Testing)
- Version of Next.js: 9.1.2

**Comment by KhaledEmaraDev** [comment]
Update:
API routes have nothing to do with the problem. I moved the API into server.js and still had the same problem.

**Comment by Timer** [comment]
Hey there! Support requests belong on our community Spectrum.

https://spectrum.chat/next-js

Thanks!

**Comment by KhaledEmaraDev** [comment]
@Timer I was under the impression that this is a bug in next.js.

I followed the docs every step of the way to create this reproducible example. I made sure to check the examples in this repository. I think the code I have here closely resembles one of the examples there.

As I mentioned in the original post, this works for 9 out of 10 scenarios. So, I don't think I made a mistake in the code here. It works under development mode; just after one minute. It works on desktop, both production and development.

FWIW, I have a post on [spectrum](https://spectrum.chat/next-js/general/next-link-doesnt-work-with-custom-server-routing-if-using-api-routes~367896cb-2f3d-42ea-b00f-637356c6da24). Which hasn't received any feedback. Not sure If I should post, again.
